### ISSUE TITLE
What's the algorithm NVDA uses to decide whether the focus should be placed on the same item after a page refresh?

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4998]
**Reported by jmuheim on 2015-03-25 20:12**
Hello

I'm developing a web application. When somebody tries to create a new blog post, and there are validation errors, the page is displayed again, with error messages. I want to set the focus directly into the first invalid input using JavaScript, but NVDA most of the time decides against this and focuses the last focused element, e.g. the "submit" button of the form (which definitely is not what the user wants to focus first when he needs to fix the input errors).

So I'd like to know how NVDA decides whether it is on the same page as before, and whether the last focused element is still there. I'm looking for ways around this so I am able to set the focus manually.

Thanks a lot,
Josh

**Comment by ehollig** [comment]
The provided example no longer exists. I got error 503. Also, as stated in https://github.com/nvaccess/nvda/issues/4998#issuecomment-155335559, the program is working properly. Closing as worksforme
